PokeAlarm/Manager.py

```diff
@@ -740,6 +796,185 @@ def process_gym(self, gym):
         for thread in threads:
             thread.join()
 
+    def process_egg(self, egg):
+        # Quick check for enabled
+        if self.__egg_settings['enabled'] is False:
+            log.debug("Egg ignored: notifications are disabled.")
+            return
+
+        id_ = egg['id']
+
+        raid_end = egg['raid_end']
+
+        # raid history will contains any raid processed
+        if id_ in self.__raid_hist:
+            old_raid_end = self.__raid_hist[id_]['raid_end']
+            if old_raid_end == raid_end:
+                if self.__quiet is False:
+                    log.info("Raid {} ignored. Was previously processed.".format(id_))
+                return
+
+        self.__raid_hist[id_] = dict(raid_end=raid_end, pkmn_id=0)
+
+        # don't alert about expired raids
+        if datetime.utcnow() > raid_end:
```

Comment:
This should really use time_left, similar to pokemon to allow users to not receive notifications if there isn't enough time left remaining.